File: SDKs/youneedabudget.com/1.0.0/python/src/sdk/models/operations/getpayeebyid.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class GetPayeeByIDPathParams:
    budget_id: str = field(metadata={'path_param': { 'field_name': 'budget_id', 'style': 'simple', 'explode': False }})
    payee_id: str = field(metadata={'path_param': { 'field_name': 'payee_id', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetPayeeByIDRequest:
    path_params: GetPayeeByIDPathParams = field()
    

@dataclass
class GetPayeeByIDResponse:
    content_type: str = field()
    status_code: int = field()
    error_response: Optional[shared.ErrorResponse] = field(default=None)
    payee_response: Optional[shared.PayeeResponse] = field(default=None)
    

File: SDKs/youneedabudget.com/1.0.0/typescript/src/sdk/models/operations/getpayeebyid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetPayeeByIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=budget_id" })
  budgetId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=payee_id" })
  payeeId: string;
}


export class GetPayeeByIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetPayeeByIdPathParams;
}


export class GetPayeeByIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  errorResponse?: shared.ErrorResponse;

  @SpeakeasyMetadata()
  payeeResponse?: shared.PayeeResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}
